services/metadata_service/api/artifact.py

```diff
@@ -137,7 +137,7 @@ async def get_artifacts_by_task(self, request):
         )
```

Comment:
`self._async_table.get_artifact_in_task(...)` returns a `DBResponse` with `.body` attribute of type string and a `.response_code` of 500, indicating an error. The string-type body is set in `db_utils.aiopg_exception_handling`.

`ArtifactsApi._filter_artifacts_by_attempt_id(...)` expects `artifacts.body` to be a list, which is what `get_artifact_in_task` returns when no error has occurred (and `fetch_single==False`, which is the case in `get_artifact_in_task`). In the event of no error, `.response_code == 200`.